closed/adds/jdk/src/share/native/jdk/crypto/jniprovider/NativeCrypto.c

```diff
@@ -162,19 +165,31 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_DigestComputeAnd
     unsigned char* digestNative;
 
     if (message != NULL) {
-       messageNative = (*env)->GetPrimitiveArrayCritical(env, message, &isCopy);
-       messageNative = messageNative + messageOffset;
-       if (1 != EVP_DigestUpdate(context->ctx, messageNative, messageLen)) handleErrors();
-           (*env)->ReleasePrimitiveArrayCritical(env, message, NULL, 0);
+        messageNative = (*env)->GetPrimitiveArrayCritical(env, message, &isCopy);
+
+        if (messageNative == NULL)
+            return -1;
+
+        messageNative = messageNative + messageOffset;
+
+        if (1 != EVP_DigestUpdate(context->ctx, messageNative, messageLen))
+            handleErrors();
+
+        (*env)->ReleasePrimitiveArrayCritical(env, message, messageNative, 0);
     }
 
     digestNative = (*env)->GetPrimitiveArrayCritical(env, digest , &isCopy);
+
```

Comment:
Above line will not compile as isCopy is removed.

Reply:
isCopy is not removed for this method. But, it has to be removed as we do not rely on this flag.